### PR TITLE
Add `backup` tool

### PR DESCRIPTION
**Pull request by loheagn**
The PR contains two main changes:

1. Enhance the `restore` tool. The `restore` tool can print the log of the `terraform apply` now.

2. Add the `backup` tool.

### CODE REVIEW

**Comment by loheagn** [comment]
Fix #310

File: hack/tool/backup_restore/README.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ This go module is a command line tool to back up and restore the configuration a
...
+
+The main usage of the `backup` subcommand is:
+
+```shell
+go main.go backup --name <name of the Configuration> --namespace <namespace of the Configuration>
+```
+
+Then you will get the `cofiguration.yaml` and the `state.json` in the workdir.
+
+Next, you can restore the Configuration and the Terraform state in another Kubernetes cluster using the `restore` subcommand.
```
Comment (zzxwill) [inline hack/tool/backup_restore/README.md:19]: The configuration could not get restored as some fields of metadata need to be cleaned before restoring.

Comment (loheagn) [reply hack/tool/backup_restore/README.md:19]: Actually, the configuration will be cleaned before encode it into yaml.

You can checkout the `cleanUpConfiguration` function [here](https://github.com/kubevela/terraform-controller/pull/315/files#diff-ef92b8134a03fc52abb41ba26a23883a872368e9cfb5d993395a328e3f0c9a8eR65).

**Review by zzxwill — APPROVED** [review]
Great job!

**Pull request merged**